src/Compatibility.StorageTable.AcceptanceTests/SetupFixture.cs

```diff
@@ -18,42 +19,30 @@ public async Task OneTimeSetUp()
             var account = CloudStorageAccount.Parse(connectionString);
             TableClient = account.CreateCloudTableClient();
 
-            allSagaDataTypeNames = GetType().Assembly.GetTypes().Where(x => x.GetInterfaces().Contains(typeof(IContainSagaData)))
-                .Select(x => x.Name).ToArray();
+            TablePrefix = $"{Path.GetFileNameWithoutExtension(Path.GetTempFileName())}{DateTime.UtcNow.Ticks}".ToLowerInvariant();
 
-            foreach (var dataTypeName in allSagaDataTypeNames)
+            allConventionalSagaTableNamesWithPrefix = GetType().Assembly.GetTypes().Where(x => x.GetInterfaces().Contains(typeof(IContainSagaData)))
+                .Select(x => $"{TablePrefix}{x.Name}").ToArray();
+
+            foreach (var tableName in allConventionalSagaTableNamesWithPrefix)
             {
-                var table = TableClient.GetTableReference(dataTypeName);
+                var table = TableClient.GetTableReference(tableName);
                 await table.CreateIfNotExistsAsync();
             }
         }
 
         [OneTimeTearDown]
-        public Task OneTimeTearDown()
-        {
-            // we can't delete the tables due to conflicts when recreating
-            return Task.CompletedTask;
-        }
-
-        public static string GetEnvConfiguredConnectionStringForPersistence()
+        public async Task OneTimeTearDown()
         {
-            var environmentVartiableName = "AzureStoragePersistence_ConnectionString";
-            var connectionString = GetEnvironmentVariable(environmentVartiableName);
-            if (string.IsNullOrEmpty(connectionString))
+            foreach (var tableName in allConventionalSagaTableNamesWithPrefix)
             {
-                throw new Exception($"Oh no! We couldn't find an environment variable '{environmentVartiableName}' with Azure Storage connection string.");
+                var table = TableClient.GetTableReference(tableName);
+                await table.DeleteIfExistsAsync();
```

Comment:
Maybe when all here

Reply:
Yeah, I was wondering the same as you could have a collection of delete tasks returned into Task.WhenAll, no?